packages/cardpay-sdk/sdk/utils/merchant.ts

```diff
@@ -0,0 +1,45 @@
+import { isAddress } from 'web3-utils';
+
+export function validateMerchantId(value: string) {
+  if (!value.trim().length) return 'This field is required';
+  if (/[^a-z0-9]/.test(value))
+    return 'The Merchant ID can only contain lowercase letters or numbers, no special characters';
+  else if (value.length > 50)
+    return `The Merchant ID cannot be more than 50 characters long. It is currently ${value.length} characters long`;
+  return '';
+}
+
+interface MerchantPaymentURLParams {
+  domain?: string;
+  merchantSafeID: string;
+  amount?: number;
+  network: string;
+  currency?: string;
+}
+
+export const generateMerchantPaymentUrl = ({
+  domain = 'cardwallet:/',
+  merchantSafeID,
+  amount,
+  network,
+  currency = 'SPD',
+}: MerchantPaymentURLParams) => {
+  const handleAmount = amount ? `amount=${amount}&` : '';
+
+  return `${domain}/pay/${network}/${merchantSafeID}?${handleAmount}currency=${currency}`;
+};
+
+export const isValidMerchantPaymentUrl = (merchantPaymentUrl: string) => {
+  let url = new URL(merchantPaymentUrl);
+  if (url.protocol !== 'cardwallet:' || url.hostname) {
```

Comment:
@Aierie isn't it should be like `if (url.protocol !== 'cardwallet:' || !url.hostname) {`?
just wonder you missded `!` before `url.hostname`

Reply:
I think we're not expecting a hostname. However, I realised upon trying this in both the browser and node environments that node and browser implementations of `URL` are different.

For:

```
new URL('cardwallet://pay/sokol/0xE73604fC1724a50CEcBC1096d4229b81aF117c94').hostname
```

Browser returns "", node returns "pay". I'll remove this check